forge/src/runner.rs

```diff
@@ -258,6 +292,28 @@ impl<'a, S: Clone, E: Evm<S>> ContractRunner<'a, S, E> {
                 }
             },
         };
+
+        if let Some(evm_traces) = self.evm.traces() {
+            if !evm_traces.is_empty() && self.evm.tracing_enabled() {
+                let trace = evm_traces.into_iter().next().expect("no trace");
+                trace.update_identified(
+                    0,
+                    known_contracts.expect("traces enabled but no identified_contracts"),
+                    &mut ident,
+                    self.evm,
+                );
+                identified_contracts = Some(ident);
+                if let Some(ref mut traces) = traces {
+                    traces.push(trace);
+                } else {
+                    // we didnt have a setup
+                    traces = Some(vec![trace]);
+                }
+            }
+        }
```

Comment:
Why do you need to do this twice, instead of having the setUp and calls logs in the same traces?

Reply:
answer: i probably dont, but setup may not exist. Which is why i did it that way, and there is a dumb thing where there is an empty trace that sticks around for some reason, thus the .next